src/main/java/com/checkmarx/flow/service/ResultsService.java

```diff
@@ -191,12 +191,50 @@ private void handleJiraCase(ScanRequest request, ScanResults results) throws Jir
             log.info("Processing results with JIRA issue tracking");
             jiraService.process(results, request);
         } catch (RestClientException e) {
+            Map<String, ScanResults.XIssue> nonPublishedScanResultsMap = jiraService.getNonPublishedScanResults();
             if (e.getStatusCode().isPresent() && e.getStatusCode().get() ==  HttpStatus.NOT_FOUND.value()) {
                 throw new JiraClientRunTimeException("Jira service is not accessible for URL: " + jiraService.getJiraURI(), e);
+            } else if (e.getStatusCode().isPresent() &&
```

Comment:
Consider moving the below line to be executer after the else and before the if

Map<String, ScanResults.XIssue> nonPublishedScanResultsMap = jiraService.getNonPublishedScanResults();